ledger/eval/eval_test.go

```diff
@@ -1445,3 +1401,179 @@ func TestExpiredAccountGeneration(t *testing.T) {
 	require.Equal(t, crypto.VRFVerifier{}, recvAcct.SelectionID)
 	require.Equal(t, merklesignature.Verifier{}.Commitment, recvAcct.StateProofID)
 }
+
+func TestEval_EndOfBlockStake(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	proto := protocol.ConsensusFuture
+	mainnetDigest, digestError := crypto.DigestFromString("YBQ4JWH4DW655UWXMBF6IVUOH5WQIGMHVQ333ZFWEC22WOJERLPQ")
+	require.NoError(t, digestError)
+
+	var tests = []struct {
+		actual      basics.MicroAlgos
+		expected    basics.MicroAlgos
+		round       basics.Round
+		genesisHash crypto.Digest
+		err         string
+	}{
+		// Normal round mainnet, matched stake => no error
```

Comment:
why mainnet?

Reply:
talked offline, this  has explicit round numbers to actually hit newly added switch-case for mainnet.